### PR TITLE
22/history carousel

### LINKED ISSUE #22: 🎨 Create about us history carousel
#  Create about us history carousel
---

Quick links: [Summary](#summary) | [Description](#description) | [Technical-Analysis](#Technical-Analysis) | [Acceptance-Criteria](#Acceptance-Criteria)

---

### Summary

As a user I want a carousel that can display a history timeline of the organisation

### Description

In this section the user can track the progress of the organisation throughout the years.

Then section has the following:

1. A title
2. A timeline carousel (reuse #21 )
3. A call to action message

The timeline carousel needs to appear as shown below including navigation functionality. When a user clicks the previous or next button they should navigate accordingly with ease.

Screenshot 2023-02-02 at 20 10 15

Screenshot 2023-02-02 at 20 10 22

Your assignment is to implement the component to appear as above.

Please take note of all the design features 

#### Link to full design:
```
https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/63e63506-4ea3-4d02-af28-f7cf148b47d9

mobile 

https://xd.adobe.com/view/7b72572b-b40a-4618-bfa6-5500ab28f13b-2e21/screen/0efb593f-dc28-4e6d-86f3-098699455ad4
```
### Technical Analysis

This component will be created in  libs/pages/elewa/about-us

### Acceptance Criteria

1. When rendered the component is pixel perfect compared to the design above.
6. The component is clean and does not have more than 50 lines
7. The component SCSS uses CSS variables for colour, font, ... Connected to the main style.scss
8. The code is isolated to the libs/page/home package, unless a justified reason is specified.
9. The component is aligned using CSS flex
10. The design is responsive for screen, tablet and phone
11. The image renders fast and has multiple srcsets

### PR DESCRIPTION
**Pull request by munenebrian**
# Description
Added the title and call to action message and implemented the previously created carousel.

Fixes #22 

## Type of change

- [ ] New feature (non-breaking change which adds functionality)

# Screenshot (optional)

# Checklist:
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: libs/pages/elewa/about-us/src/lib/components/history-carousel/history-carousel.component.html [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+        <p class="count-text ">Careers launched</p>
+    </div>
+    <div class="counter-col">
+        <h2>{{projectcount}}+</h2>
+        <p class="count-text ">Projects finished</p>
+    </div>
+    <div class="counter-col">
+        <h2>{{investorcount}} </h2>
+        <p class="count-text ">Investors</p>
+    </div>
```
Comment (LemmyMwaura) [inline libs/pages/elewa/about-us/src/lib/components/history-carousel/history-carousel.component.html:17]: This is a completely different issue and also the component name does not match the contents and vice versa.

File: libs/pages/elewa/about-us/src/lib/pages/about-us-page/about-us-page.component.html [hunk 2]
```diff
@@ -5,6 +5,10 @@
 
       <elewa-group-about-us-culture></elewa-group-about-us-culture>
       <elewa-group-team-members-carousel></elewa-group-team-members-carousel>
+      <elewa-group-history-carousel></elewa-group-history-carousel>
```
Comment (LemmyMwaura) [inline libs/pages/elewa/about-us/src/lib/pages/about-us-page/about-us-page.component.html:8]: Also here, two history carousels ??

**Review by LemmyMwaura — CHANGES_REQUESTED** [review]

File: libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts [hunk 3]
```diff
@@ -19,6 +19,10 @@ import { NextDirective } from './directives/next.directive';
 import { PrevDirective } from './directives/prev.directive';
 
 import { AboutUsRoutingModule } from './about-us.routing';
+import { HistoryCarouselComponent } from './components/history-carousel/history-carousel.component';
+import { ElewaGroupImageAndTextBannerComponent } from 'libs/features/components/banners/src/lib/banners/elewa-group-image-and-text-banner/elewa-group-image-and-text-banner.component';
```
Comment (LemmyMwaura) [inline libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts:23]: Import from the aliased path

File: libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts [hunk 4]
```diff
@@ -19,6 +19,10 @@ import { NextDirective } from './directives/next.directive';
...
+import { AboutUsHistoryCarouselComponent } from './components/about-us-history-carousel/about-us-history-carousel.component';
+import { ElewaHorizontalTimelineCarouselComponent } from 'libs/features/components/ui-lists/src/lib/elewa-horizontal-timeline-carousel/elewa-horizontal-timeline-carousel.component';
```
Comment (LemmyMwaura) [inline libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts:25]: Same here, import from aliased path

File: libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts [hunk 5]
```diff
@@ -34,6 +38,12 @@ import { AboutUsRoutingModule } from './about-us.routing';
     PrevDirective,
     AboutUsCultureComponent,
     AboutUsPageComponent,
+    HistoryCarouselComponent,
+    ElewaGroupImageAndTextBannerComponent,
+    AboutUsHistoryCarouselComponent,
```
Comment (LemmyMwaura) [inline libs/pages/elewa/about-us/src/lib/elewa-about-us.module.ts:43]: This component (ELewaHorizontal) does not belong to this module

File: libs/features/components/ui-lists/src/lib/elewa-horizontal-timeline-carousel/elewa-horizontal-timeline-carousel.component.ts [hunk 6]
```diff
@@ -1,3 +1,5 @@
+/* eslint-disable @typescript-eslint/no-empty-function */
+/* eslint-disable @typescript-eslint/no-inferrable-types */
```
Comment (IanOdhiambo9) [inline libs/features/components/ui-lists/src/lib/elewa-horizontal-timeline-carousel/elewa-horizontal-timeline-carousel.component.ts:2]: remove

File: apps/elewa-group-website/src/_variables.scss [hunk 7]
```diff
@@ -72,4 +72,13 @@
...
   --elewa-group-website-timeline-header: 38px;
+
+  // about-history
+  --elewa-group-website-container-background: rgba(128, 128, 128, 0.329);
+  --elewa--group-website-title-font-size: 40px;
+  --elewa--group-website-action-message-font-size: 40px;
+  --elewa--group-website-action-message-font-weight: 100;
+  --elewa--group-website-title-font-size-media: 30px;
+  --elewa--group-website-action-message-font-size-media: 30px;
+  --elewa--group-website-action-message-font-weight-media: 100;
```
Comment (IanOdhiambo9) [inline apps/elewa-group-website/src/_variables.scss:83]: Not necessary here, just put this styles in the component

File: libs/pages/elewa/about-us/src/lib/components/history-carousel/history-carousel.component.html [hunk 8]
```diff
@@ -0,0 +1,18 @@
+<div class="wrapper">
+    <div class="counter-col">
+        <h2>{{peoplecount}}+</h2>
+        <p class="count-text ">People trained</p>
+    </div>
+    <div class="counter-col">
+        <h2>{{careercount}}k</h2>
```
Comment (IanOdhiambo9) [inline libs/pages/elewa/about-us/src/lib/components/history-carousel/history-carousel.component.html:7]: there's two carousels here??

**Review by IanOdhiambo9 — CHANGES_REQUESTED** [review]

**Comment by IanOdhiambo9** [comment]
This issue has been completed

**Pull request closed without merging**